### PR TITLE
Dev-docs: Git and coding standards-related pages

### PR DESCRIPTION
**Pull request by josephframsay**
Pages associated with Git, GitHub and coding standards were moved to the new `dev-docs` target

**To test:**

Build the docs and have a quick read through them to make sure everything looks correct and sounds sensible

Refs #22082 

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: dev-docs/source/GitWorkflow.rst [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+to drop the prefix.
+
+     
+.. _dev-docs-git-workflow-pull-requests:
+   
+Pull Requests
+^^^^^^^^^^^^^
+
+For an general overview of using pull requests on GitHub look `here
+<https://help.github.com/articles/using-pull-requests/>`__
```
Comment (DanNixon) [inline dev-docs/source/GitWorkflow.rst]: Full stop at end of line.

File: dev-docs/source/GitWorkflow.rst [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+  to read on the overview page
+
+  - The title should **not** contain the issue number
+- `Reference the issue which the pull request is closing  <https://github.com/blog/1506-closing-issues-via-pull-requests>`_, using one of `these <https://help.github.com/articles/closing-issues-via-commit-messages>`_ keywords
+- Ensure the description follows the format described by the `PR
+  template
+  <https://github.com/mantidproject/mantid/blob/master/.github/PULL_REQUEST_TEMPLATE.md>`_
+  on GitHub
+
+A good example is `here <https://github.com/mantidproject/mantid/pull/18713>`__
```
Comment (DanNixon) [inline dev-docs/source/GitWorkflow.rst]: `.`

File: dev-docs/source/GitWorkflow.rst [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+     to GitHub - the pull request will update automatically
+
+.. _dev-docs-git-workflow-public-private-branches:
+
+Public and Private Branches
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When naming `public branches
+<http://github.com/mantidproject/mantid/branches>`_ that will be
+synchronized with GitHub, please follow the convention of
```
Comment (DanNixon) [inline dev-docs/source/GitWorkflow.rst]: `synchronized with` -> `pushed to`

Synchronization is not a thing it Git.

File: dev-docs/source/IssueTracking.rst [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+- Code changes required: generate a **pull request**
+  (:ref:`dev-docs-git-workflow-pull-requests`)
+
+  - Make sure the issue is referenced with ``#issue-number``. It will
+    then be closed when the pull request is merged
+- No code changes required: Close the issue, with a comment explaining
+  why the issue should be closed, for example:
+
+  - This is a duplicate of #1234
```
Comment (DanNixon) [inline dev-docs/source/IssueTracking.rst]: May as well add umbrella issues to this list as they often are not closed by a single PR

File: dev-docs/source/IssueTracking.rst [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+.. _dev-docs-issue-tracking-labels:
+     
+Labels
+^^^^^^
+
+GitHub allows you to assign various labels to your issues and pull
+requests. Labelling your issues and pull requests adequately will
+ensure that they are picked up by the right people.
+
+Labels fall into several groups: (*please note that 'ticket' is used
```
Comment (DanNixon) [inline dev-docs/source/IssueTracking.rst]: This asterisk needs escaping, it is being rendered as a link.

**Review by DanNixon — CHANGES_REQUESTED** [review]
All looks good, just a few comments.

File: dev-docs/source/CodingStandards/UnitTestStandards.rst [hunk 6]
```diff
@@ -0,0 +1,23 @@
+==========================
+Unit Test Coding standards
+==========================
+
+The code for unit test classes should follow the standards as set out
+in :ref:`dev-docs-cpp-standards` with the following differences.
+
+- The file and class name should be of the form "ClassNameTest", for
+  example WorkspaceTest, LoggerTest.
```
[the remark this answers is not in the document]
Comment (josephframsay) [reply dev-docs/source/CodingStandards/UnitTestStandards.rst]: @DanNixon could you explain what you mean here? I didn't realise you could have class names that aren't raw strings?

[the remark this answers is not in the document]
Comment (DanNixon) [reply dev-docs/source/CodingStandards/UnitTestStandards.rst]: Sorry I mean the inline monospace style (I usually refer to it as "raw text").

[the remark this answers is not in the document]
Comment (josephframsay) [reply dev-docs/source/CodingStandards/UnitTestStandards.rst]: Oh yep, sorry - that makes sense!

**Review by DanNixon — APPROVED** [review]

**Pull request merged**